### PR TITLE
GEOMESA-889 Make exception in getRecommendedBitsResolutionForPolygon …

### PR DESCRIPTION
**Pull request by apc3n**
…a class for case matching

### CODE REVIEW

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geohash/GeohashUtils.scala [hunk 1]
```diff
@@ -521,12 +520,14 @@ object GeohashUtils
...
 
       if (numBoxes==0)
-        throw new Exception("Could not satisfy constraints, resolutions " + resolutions + ", constraints " + constraints + ".")
+        throw UnsatisfiedConstraintsException(s"Could not satisfy constraints, resolutions $resolutions, constraints $constraints.")
 
       RecommendedResolution(maxBits, numBoxes)
     }
   }
 
+  case class UnsatisfiedConstraintsException(message: String) extends Exception(message)
```
Comment (jnh5y) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geohash/GeohashUtils.scala]: @apc3n @cne1x Would an IllegalArgumentException here be better/fine?

Comment (apc3n) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geohash/GeohashUtils.scala]: @jnh5y We don't actually need the case class for case matching anymore. So we can either get rid of this PR all together or use it as an opportunity to change this to an illegal argument exception. Let me know what you think is best.

Comment (anthonyccri) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geohash/GeohashUtils.scala]: Go with IllegalArgumentException - it's more standard.

**Comment by elahrvivaz** [comment]
4c5e20d615cd9450b2f630143080ea3e4763f502

**Pull request closed without merging**